monai/networks/nets/mednext.py

```diff
@@ -0,0 +1,244 @@
+                    dim=spatial_dims_str,
+                    grn=grn,
+                )
+            )
+
+            dec_stages.append(
+                nn.Sequential(
+                    *[
+                        MedNeXtBlock(
+                            in_channels=init_filters * (2 ** (len(blocks_up) - i - 1)),
+                            out_channels=init_filters * (2 ** (len(blocks_up) - i - 1)),
+                            exp_r=dec_exp_r[i],
+                            kernel_size=dec_kernel_size,
+                            do_res=do_res,
+                            norm_type=norm_type,
+                            dim=spatial_dims_str,
+                            grn=grn,
+                        )
+                        for _ in range(num_blocks)
+                    ]
+                )
+            )
+
+        self.up_blocks = nn.ModuleList(up_blocks)
+        self.dec_stages = nn.ModuleList(dec_stages)
+
+        self.out_0 = OutBlock(in_channels=init_filters, n_classes=out_channels, dim=spatial_dims_str)
+
+        if deep_supervision:
+            out_blocks = [
+                OutBlock(in_channels=init_filters * (2**i), n_classes=out_channels, dim=spatial_dims_str)
+                for i in range(1, len(blocks_up) + 1)
+            ]
+
+            out_blocks.reverse()
+            self.out_blocks = nn.ModuleList(out_blocks)
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor | list[torch.Tensor]:
+        """
+        Forward pass of the MedNeXt model.
+
+        This method performs the forward pass through the model, including:
+        - Stem convolution
+        - Encoder stages and downsampling
+        - Bottleneck blocks
+        - Decoder stages and upsampling with skip connections
+        - Output blocks for deep supervision (if enabled)
+
+        Args:
+            x (torch.Tensor): Input tensor.
+
+        Returns:
+            torch.Tensor or list[torch.Tensor]: Output tensor(s).
+        """
+        # Apply stem convolution
+        x = self.stem(x)
+
+        # Encoder forward pass
+        enc_outputs = []
+        for enc_stage, down_block in zip(self.enc_stages, self.down_blocks):
+            x = enc_stage(x)
+            enc_outputs.append(x)
+            x = down_block(x)
+
+        # Bottleneck forward pass
+        x = self.bottleneck(x)
+
+        # Initialize deep supervision outputs if enabled
+        if self.do_ds:
+            ds_outputs = []
+
+        # Decoder forward pass with skip connections
+        for i, (up_block, dec_stage) in enumerate(zip(self.up_blocks, self.dec_stages)):
+            if self.do_ds and i < len(self.out_blocks):
+                ds_outputs.append(self.out_blocks[i](x))
+
+            x = up_block(x)
+            x = x + enc_outputs[-(i + 1)]
+            x = dec_stage(x)
+
+        # Final output block
+        x = self.out_0(x)
+
+        # Return output(s)
+        if self.do_ds and self.training:
+            return (x, *ds_outputs[::-1])
```

Comment:
since we're changing the output type here anyway, a dict could be less error-prone here

Reply:
In this case, the output is required to be a list for subsequent use in losses like https://docs.monai.io/en/stable/losses.html#monai.losses.DeepSupervisionLoss